app/controllers/air_zones_controller.rb

```diff
@@ -9,6 +9,7 @@ class AirZonesController < ApplicationController
   # 422 HTTP status from API means vehicle data incomplete so the compliance calculation is not possible.
   rescue_from BaseApi::Error422Exception, with: :unable_to_determine_compliance
   # checks if vrn is present in the session
+  before_action :encrypt_vrn, only: :compliance
   before_action :check_vrn
```

Comment:
```suggestion
  before_action :check_vrn
  before_action :encrypt_vrn, only: :compliance
```
global methods should be always first, both cases will work properly, but logic should same as methods execution.